src/test/server/test_dispatcher.py

```diff
@@ -0,0 +1,125 @@
+from copy import deepcopy
+from unittest import TestCase
+
+from ja.common.job import Job, JobStatus
+from ja.server.database.types.job_entry import DatabaseJobEntry
+from ja.server.database.types.work_machine import WorkMachine
+from ja.server.dispatcher.dispatcher import Dispatcher
+from test.proxy.worker_proxy_factory import WorkerProxyDummyFactory
+
+
+class TestDispatcher(TestCase):
+
+    def _assert_distribution_correct_a(self) -> None:
+        # Test if jobs are dispatched correctly by invoking the proxies:
```

Comment:
Why do we need to test this in an indirect way here?
We control the factory and we know it returns mock objects. We can directly access those mock objects and check the job status, right?

Reply:
I implemented tests using only the public interface so that we can directly compare the results of mock implementations and the actual implementations during the integration phase.